indexer/packages/postgres/src/stores/fill-table.ts

```diff
@@ -156,11 +158,40 @@ export async function findAll(
     Ordering.DESC,
   );
 
-  if (limit !== undefined) {
+  if (limit !== undefined && page === undefined) {
     baseQuery = baseQuery.limit(limit);
   }
 
-  return baseQuery.returning('*');
+  /**
+   * If a query is made using a page number, then the limit property is used as 'page limit'
+   */
+  if (page !== undefined && limit !== undefined) {
+    /**
+     * We make sure that the page number is always >= 1
+     */
+    const currentPage = Math.max(1, page);
+    const offset = (currentPage - 1) * limit;
+
+    /**
+     * We need to remove the sorting as it is not necessary in this case.
+     * Also a casting of the ts type is required since the infer of the type
+     * obtained from the count is not performed.
+     */
+    const count = await baseQuery.clone().clearOrder().count({ count: '*' }).first() as unknown as { count?: string };
```

Comment:
What's the sort order in this case then? Is there some default?

Reply:
Here we're just counting all the elements. No matter what order they are in, because the COUNT() function returns the number of lines that match a given criterion. For this reason we get the first element as it is the only one available.

Reply:
Clients are expecting a reasonable default order per endpoint. That way, random records won't be skipped/repeated in subsequent pages. I think it makes sense to use ascending `eventId` order for trades/fills/transfers related endpoints, and `orderId` for the orders endpoint.

Reply:
I think there may be a misunderstanding here. The sort order is only removed for a query that asks for a count (a total number of rows), not for the list of items. 
Such a query is used to retrieve the total number of items, which we return to the client to say "there are x rows in total". This helps the user to know how many records have not yet been received and for which they could make requests.
Regarding the list of items, as you can see, the order is maintained because the `clone()` method _clones the current query chain, which is useful for reusing partial query snippets in other queries without mutating the original_.

I hope this helps to clarify things.

Reply:
I understand. I don't think the change is required in this file.

But can you pass in an orderBy param set to `eventId` ASC in fills-controller, trades-controller, and transfers-controller? And can you add unit tests to those controllers verifying the order? We also want pagination on the orders endpoint too.

Reply:
Got it 💪

In such a scenario, I think you might agree that adding the ordering feature is not strictly related to the pagination feature. So it might be interesting (and probably better) to add such a feature, **but via a new issue and pull request to keep the context**.

WRT the ability to add pagination on the orders endpoint too, that might be a good suggestion.

We'll work on it 👷

Reply:
If that's the case, can you remove all the changes to the API in this PR? We don't want to be in a partial state of having pagination exposed but not having the correct order.

Reply:
Yeah. We will remove the "controller level" changes and leave only the "table level" changes.
Then we'll move to a new PR related to the same issue (since you want to keep the pagination and sorting related) to add both pagination and ordering at "controller level" and we can merge the two.

Do you agree?

Reply:
Agreed, thank you!

Reply:
As agreed, we have implemented the changes at table level. 
We're now moving to the controller level with a new PR!